### PR TITLE
email 2fa

### PR DESCRIPTION
**Pull request by ComputerGenius152**
Closes https://github.com/jdabtieu/CTFOJ/issues/40

### CODE REVIEW

File: application.py [hunk 1]
```diff
@@ -276,6 +299,30 @@ def confirm_register(token):
...
+                               site_key=app.config['HCAPTCHA_SITE'])
+
+    # Log user in
+    user = db.execute(
+        "SELECT * FROM users WHERE email = :email", email=token['email'])[0]
+    session["user_id"] = user["id"]
+    session["username"] = user["username"]
+    session["admin"] = False
+
+    return redirect("/problem/helloworld")
```
Comment (jdabtieu) [inline application.py:325]: should redirect to / after logging in. People are only redirected to /problem/helloworld after registering

**Review by jdabtieu — CHANGES_REQUESTED** [review]
How does a user toggle 2FA? I'm assuming we'll combine this & change password in the navbar to something like user settings, and allow people to change it there?

Anyways, see my other comment for the change requested.

File: application.py [hunk 2]
```diff
@@ -276,6 +299,30 @@ def confirm_register(token):
...
+        return render_template("login.html",
+                               message="Login verification link expired; Please re-login",
+                               site_key=app.config['HCAPTCHA_SITE'])
+
+    # Log user in
+    user = db.execute(
+        "SELECT * FROM users WHERE email = :email", email=token['email'])[0]
+    session["user_id"] = user["id"]
+    session["username"] = user["username"]
+    session["admin"] = False
```
Comment (jdabtieu) [inline application.py:323]: this should be user["admin"], not False

**Comment by ComputerGenius152** [comment]
> How does a user toggle 2FA? I'm assuming we'll combine this & change password in the navbar to something like user settings, and allow people to change it there?

Right now, 2fa can only be manually set via the database. I'm now working on a settings page for password and 2fa changes.

**Pull request merged**